### DISCUSSION TITLE
Doorbell notification not working properly

### DISCUSSION THREAD

**Discussion by Onionking18** [HomeSpan/HomeSpan#715] [Q&A]
Hi everyone! First of all I want to give my respect for this library, it made my life easier. 
Lately I encountered one issue when I tried to automate my Intercom. It is not working properly when it comes to the notification. I used an ESP32 DEV kit and tried to use it as an analog voltage reader. When the tension is above 1.5 volts I should be notified. It seems that is not working as intended and I think I might get some answers from here if anyone is willing to help. 
[Intercom_final.zip](https://github.com/HomeSpan/HomeSpan/files/13689248/Intercom_final.zip)

`#include "Arduino.h"

#include "HomeSpan.h" 
#include "doorbell.h"
#include "DEV_open.h"

int doorbellPin = 32;
const int triggerPin = 33;
const int doorbellAnPin = 34;

void setup() {
  // put your setup code here, to run once:
  Serial.begin(115200);
  pinMode (triggerPin, OUTPUT);
  pinMode (doorbellAnPin, INPUT);
  digitalWrite(triggerPin, HIGH);
  homeSpan.setPairingCode("11122333");
  homeSpan.setQRID("111-22-333");

  homeSpan.begin(Category::ProgrammableSwitches, "Doorbell");
       
  new SpanAccessory();
    new Service::AccessoryInformation();    
      new Characteristic::Identify();
    new Doorbell(doorbellPin); 

  new SpanAccessory();
    new Service::AccessoryInformation();   
      new Characteristic::Identify();                       
    new DEV_open(18,19);  
}

bool flag=false;
uint32_t indx=0;
int32_t adcValue=0;

void loop() {
  adcValue=analogRead(doorbellAnPin);
  // Reading potentiometer value
  if (adcValue>2048){
    if (flag==false){
      Serial.println(adcValue);
      digitalWrite(triggerPin, LOW);
      Serial.print(indx);
      indx++;
      Serial.println("LOW");
    }
    flag=true;
  } else {
    if (flag==true){
      Serial.println(adcValue);
      digitalWrite(triggerPin, HIGH);
      Serial.print(indx);
      indx++;
      Serial.println("HIGH");
      
    }
    flag=false;
  }

  homeSpan.poll();
  
}`

`#include "HomeSpan.h"

// https://github.com/HomeSpan/HomeSpan/blob/master/examples/16-ProgrammableSwitches/16-ProgrammableSwitches.ino
struct Doorbell : Service::Doorbell // StatelessProgrammableSwitch
{
  SpanCharacteristic *switchEvent;

  Doorbell(int doorbellPin) : Service::Doorbell() // StatelessProgrammableSwitch
  {
    switchEvent = new Characteristic::ProgrammableSwitchEvent();
    new SpanButton(doorbellPin);

    LOG0("-----SETTING DOORBELL PIN INPUT ------\n");
  }

  void button(int pin, int pressType) override
  {
    LOG0("------DOORBEL PIN INPUT STATE CHANGE -------\n");
    switchEvent->setVal(pressType);

  }
};
`

**Comment by HomeSpan** [comment]
Are you using it the correct way (see [FAQ](https://github.com/HomeSpan/HomeSpan/blob/master/docs/FAQ.md#why-does-the-home-app-indicate-the-doorbell-service-is-unsupported))?
